plugins/by-name/oil-git-status/default.nix

```
@@ -0,0 +1,39 @@
+{
+  lib,
+  ...
+}:
+lib.nixvim.plugins.mkNeovimPlugin {
+  name = "oil-git-status";
+  packPathName = "oil-git-status.nvim";
+  package = "oil-git-status-nvim";
+
+  description = ''
+    Add Git Status to oil.nvim directory listings.\
+    Git status is added to the listing asynchronously after creating the oil directory\
+    listing so it won't slow oil down on big repositories. The plugin puts the status in two new sign columns\
+    the left being the status of the index, the right being the status of the working directory.
+    signcolumn needs to be configured in oil:
+    ```nix
+    plugins.oil = {
+      enable = true;
+      settings = {
+        win_options = {
+          signcolumn = "yes:2,";
+        };
+      };
+    };
+    ```
+  '';
+
+  maintainers = [ lib.maintainers.FKouhai ];
+  # Ensure oil-git-status is set up after oil is loaded
```

Nit: I'd have a blank line here:
```suggestion

  # Ensure oil-git-status is set up after oil is loaded
```